CRM/Utils/System/Joomla.php

```diff
@@ -342,6 +342,8 @@ public function authenticate($name, $password, $loadCMSBootstrap = FALSE, $realP
       }
       CRM_Utils_System::loadBootStrap($bootStrapParams, TRUE, TRUE, FALSE);
     }
+    //CRM-20904 Load all DAOs defined in hooks/extensions.
+    CRM_Core_DAO_AllCoreTables::reinitializeCache(TRUE);
```

Comment:
This is interesting.

If you said there was a bug with *Joomla extensions* where the cache in `AllCoreTables` got stale, that would make sense. (IIRC, `hook_civicrm_entityTypes` is actually a boot-critical hook -- the very hook fired. This would mean it fires before Civi has a chance to bootstrap Joomla.)

Gonna see if I can replicate...